src/java/com/puppetlabs/http/client/impl/PersistentAsyncHttpClient.java

```diff
@@ -15,13 +15,16 @@
     private CloseableHttpAsyncClient client;
     private MetricRegistry metricRegistry;
     private String metricNamespace;
+    private Boolean useURLMetrics;
 
     public PersistentAsyncHttpClient(CloseableHttpAsyncClient client,
                                      MetricRegistry metricRegistry,
-                                     String metricNamespace) {
+                                     String metricNamespace,
+                                     Boolean useURLMetrics) {
```

Comment:
I've made this change in bd4bbd1 and 38624a8 but in doing so I now get:
```
java.lang.NoSuchMethodError: com.puppetlabs.http.client.impl.PersistentAsyncHttpClient.<init>(Lorg/apache/http/impl/nio/client/CloseableHttpAsyncClient;Lcom/codahale/metrics/MetricRegistry;Ljava/lang/String;Ljava/lang/Boolean;)V
 at com.puppetlabs.http.client.Async.createClient (Async.java:27)
```

In all my tests that instantiate the Async client fail in this way, or in other tests they will fail with:

```
java.lang.NoSuchMethodError: com.puppetlabs.http.client.impl.metrics.TimerUtils.startFullResponseTimers(Lcom/codahale/metrics/MetricRegistry;Lorg/apache/http/HttpRequest;[Ljava/lang/String;Ljava/lang/String;Ljava/lang/Boolean;)Ljava/util/ArrayList;
 at com.puppetlabs.http.client.impl.JavaClient.requestWithClient (JavaClient.java:465)
```

Looking on the internet it seems that Futures require Booleans rather booleans? I don't know why it'd be failing there. I have a feeling I'm missing something super simple...